src/v/rpc/simple_protocol.cc

```diff
@@ -99,61 +99,66 @@ simple_protocol::dispatch_method_once(header h, net::server::resources rs) {
     auto fut = ctx->pr.get_future();
 
     // background!
-    (void)with_gate(rs.conn_gate(), [this, method_id, rs, ctx]() mutable {
-        auto it = std::find_if(
-          _services.begin(),
-          _services.end(),
-          [method_id](std::unique_ptr<service>& srvc) {
-              return srvc->method_from_id(method_id) != nullptr;
-          });
-        if (unlikely(it == _services.end())) {
-            rs.probe().method_not_found();
-            netbuf reply_buf;
-            reply_buf.set_status(rpc::status::method_not_found);
-            return send_reply(ctx, std::move(reply_buf)).then([ctx]() mutable {
-                ctx->signal_body_parse();
-            });
-        }
+    try {
```

Comment:
This style of exception handling will only catch errors that are emitted synchronously from with_gate.  Exceptions in the lambda inside it will result in exceptional futures returned by with_gate, so rather than an outer `try`, we need a `.handle_exception` chained on to the return value of with_gate.

(grep for `handle_exception([](const std::exception_ptr& e) {` for examples)

Reply:
use try_with_gate and handle_exception